### PR TITLE
libdill 1.6 (new formula)

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/libdill.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class Libdill < Formula
+  desc "Structured concurrency in C"
+  homepage "http://libdill.org"
+  url "https://github.com/sustrik/libdill/archive/1.6.tar.gz"
+  sha256 "2539d3440971c876b117bc87295b0caabc8c8ec5f544856ca9e74ed12705ea51"
+
```
Comment (stephengroat) [inline Formula/libdill.rb]: maybe add a link to `head` with `head "https://github.com/sustrik/libdill.git"`?

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/libdill.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
+class Libdill < Formula
+  desc "Structured concurrency in C"
+  homepage "http://libdill.org"
```
Comment (ilovezfs) [inline Formula/libdill.rb]: http://libdill.org/

File: Formula/libdill.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  url "https://github.com/sustrik/libdill/archive/1.6.tar.gz"
```
Comment (ilovezfs) [inline Formula/libdill.rb]: 1.7 is out

File: Formula/libdill.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  sha256 "2539d3440971c876b117bc87295b0caabc8c8ec5f544856ca9e74ed12705ea51"
+
+  option "without-sockets", "Compile without socket support"
```
Comment (ilovezfs) [inline Formula/libdill.rb]: let's just always build with socket support

File: Formula/libdill.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  option "without-threads", "Compile without threading support"
```
Comment (ilovezfs) [inline Formula/libdill.rb]: let's just always build with thread support

File: Formula/libdill.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  option "with-valgrind", "Compile with valgrind compatibility"
```
Comment (ilovezfs) [inline Formula/libdill.rb]: are you yourself going to use this? if not, I'd rather leave it out

File: Formula/libdill.rb [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+      int main() {
+          go(worker("Hello!"));
+          go(worker("World!"));
+          msleep(now() + 5000);
+          return 0;
+      }
+    EOS
+    system ENV.cc, "-I#{include}", "-L#{lib}", "-ldill", "-o", "test", "test.c"
+    system testpath/"test"
```
Comment (ilovezfs) [inline Formula/libdill.rb]: ./

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/libdill.rb (hunk 6 above)
Comment (alebcay) [reply Formula/libdill.rb]: Any particular reason why you'd prefer to exclude it?

File: Formula/libdill.rb (hunk 1 above)
Comment (ilovezfs) [reply Formula/libdill.rb]: let's not

File: Formula/libdill.rb (hunk 6 above)
Comment (ilovezfs) [reply Formula/libdill.rb]: because valgrind breaks during every macOS upgrade :see_no_evil:

Comment (alebcay) [reply Formula/libdill.rb]: Fair enough. [I did manage to get valgrind working on High Sierra, but nobody's said anything there yet.](https://github.com/Homebrew/homebrew-core/issues/18998#issuecomment-354189409)

**Comment by ilovezfs** [comment]
@alebcay Thanks for the PR. Happy to look at a fresh one with the feedback addressed.

File: Formula/libdill.rb (hunk 2 above)
Comment (alebcay) [reply Formula/libdill.rb]: Resolved.

**Comment by alebcay** [comment]
See #24134.

**Pull request closed without merging**